src/CharterManager.sol

```diff
@@ -195,33 +213,49 @@ contract CharterManagerImp {
         ManagedGemJoinLike(gemJoin).exit(urp, usr, val);
     }
 
-    function frob(address gemJoin, address u, address v, address w, int256 dink, int256 dart) external allowed(u) {
-        require(u == v && w == msg.sender, "CharterManager/not-matching");
-        address urp = getOrCreateProxy(u);
+    function price(bytes32 ilk, uint256 spot) internal returns (uint256 cur) {
+        (, uint256 mat) = SpotterLike(spotter).ilks(ilk);
+        cur = rmul(rmul(spot, mat), SpotterLike(spotter).par());
```

Comment:
I don't think this should be divided by the `par`. I get that this needs to be done to get the actual price, but we should be redividing this out to calculate the `tab` limit on line 242.

`require(tab <= mul(ink, rdiv(cur, _peace)), "CharterManager/below-peace-ratio");`

should be

`require(tab <= mul(ink, rdiv(rdiv(cur, par), _peace)), "CharterManager/below-peace-ratio");`

to match what is in the `Spotter` contract.

This is unnecessary though as it would just be a multiplication of `par` followed by a division. I recommend just removing the `par` multiplication altogether and leaving line 242 as-is. The price will not exactly be the price anymore, so maybe you want to adjust the name.